### DISCUSSION TITLE
Dark mode

### DISCUSSION THREAD

**Discussion by ghost** [rust-lang/crates.io#4348] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

The problem is that the default theme is a little bit too white in my opinion. Especially late at night it can be pretty dazzling I think.

**Describe the solution you'd like**

I would like there to be a dark mode/theme.

**Describe alternatives you've considered**

Making the default theme/appearance darker.

**Additional context**

I switched `main-color` and `main-bg` in the CSS and it already looked a lot better.

![image](https://user-images.githubusercontent.com/35064754/113521736-ba1f1200-959b-11eb-855b-e11066d39ad1.png)

This feature should use CSS' `@media (prefers-color-scheme: dark)`.

**Comment by softmoth** [comment]
See pull request #3585. It seems stalled due to lack of developers. In the mean time, https://darkreader.org/ helps a lot.

**Reply by Rudxain** [reply]
For security, performance, and privacy reasons (also because I like minimalism), I prefer using 1st-party solutions when possible. Chromium browsers have `about://flags/#enable-force-dark`, where "about" is a placeholder for the browser name (chrome, edge, etc...). Unfortunately, Firefox doesn't have this, so DarkReader is the only choice

**Reply by kotx** [reply]
IIRC you can use Dark Reader to generate dark theme css. This could probably be implemented more easily that way?

**Comment by Turbo87** [comment]
https://github.com/rust-lang/crates.io/pull/8559 was merged yesterday and is deployed to production now. note that by default it still shows the light mode while we're still searching for any potential issues. once everything is confirmed to work as expected we will change the default to "system".
